client/packages/common/src/ui/components/buttons/standard/DialogButton.tsx

```diff
@@ -136,6 +137,23 @@ export const DialogButton: React.FC<DialogButtonProps> = ({
 }) => {
   const t = useTranslation();
   const { variant: buttonVariant, icon, labelKey } = getButtonProps(variant);
+  const ref = React.useRef<HTMLButtonElement>(null);
+
+  const isKeyValid = (e: KeyboardEvent) => {
+    switch (variant) {
+      case 'save':
+        return e.altKey && e.code === 'KeyS';
+      case 'cancel':
+        return e.code === 'Escape';
+      default:
+        return false;
+    }
+  };
+
+  useKeyboardShortcut({
+    isKeyValid,
+    onKeyPressed: () => ref.current?.click(),
```

Comment:
What's the advantage of simulating a click, as opposed to just calling the `onClick()` function here?

If it's just a typing thing (cos the signature of `onClick` takes an event parameter), I had a quick look around and I can't see any DialogButton implementations that actually use this, so we could just remove it or make it optional?

Reply:
it's because if you do that, then the onClick becomes a captured enclosure - in the case of the save function, then it will capture the state when the dialog button is rendered, that is, it has no lines to save. 

It may be possible to resolve that and recreate the captured fn, but this is simpler. it has the side effect of removing the type inconsistency where the event parameter is required; adding that in needs a casting `as` or requires a big refactor to remove the parameter.